runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/widgets/GlobalVariableSelectWidget.tsx

```diff
@@ -12,16 +15,20 @@ interface GlobalVariableOption {
 }
 
 const GlobalVariableSelectWidget: FC<WidgetProps> = (props) => {
+    const { user } = useAuth();
+    const isAdmin = useRole([Role.ROLE_ADMIN]);
     const { globalVariables } = useSelector(globalVariableSelector);
 
-    const options = useMemo(
-        () =>
-            globalVariables.map<GlobalVariableOption>((variable) => ({
-                id: variable.id,
-                name: variable.name,
-            })),
-        [globalVariables]
-    );
+    const options = useMemo(() => {
+        const userGlobalVariables = isAdmin
+            ? globalVariables
+            : globalVariables.filter((variable) => variable.creator.id === user.id);
+
+        return userGlobalVariables.map<GlobalVariableOption>((variable) => ({
+            id: variable.id,
+            name: variable.name,
+        }));
+    }, [globalVariables]);
```

Comment:
global variables should be already filtered at this point - whole logic must be on the backend side